src/pages/UpdateServers.js

```diff
@@ -254,6 +254,10 @@ class UpdateServers extends BaseUpdateWizardPage {
   }
 
   async getServerStatuses() {
+    //we need the model from the props here, we will try to wait for them to show up
+    while(!this.props.model || !this.state.internalModel) {
+      await new Promise((resolve, reject) => setTimeout(resolve, 10));
```

Comment:
Polling for completion is an ugly way to handle this, and is going to be problematic if there is a possibility for some exception occurring that can prevent this from happening.

Since this function is called in only one place, componentDidMount, how about just passing in the model as a function parameter directly; we really don't need to even look in the state here, do we?

It looks like the `checkMonasca` also has all sorts of contortions to try to work around this same problem that could be handled in a much more straightforward way.